src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -218,23 +218,19 @@ export const getAnswerForCheckbox = (
 
 export const getAnswersForChild = (
   response: ProcessedChildrenResponse,
+  childIdx: number,
 ): ResponseFormattedForEmail[] => {
   const subFields = response.childSubFieldsArray
   if (!subFields) {
     return []
   }
-  return response.answerArray.flatMap((arr, childIdx) => {
+  return response.answerArray.flatMap((arr) => {
     // First array element is always child name
     const childName = arr[0]
     return arr.map((answer, idx) => ({
-      _id: getMyInfoChildHashKey(
-        response._id,
-        subFields[idx],
-        childIdx,
-        childName,
-      ),
+      _id: getMyInfoChildHashKey(response._id, subFields[idx], childName),
       fieldType: response.fieldType,
-      question: `Child.${subFields[idx]}`,
+      question: `Child-${childIdx + 1}.${subFields[idx]}`,
```

Comment:
Trying to understand the logic on why this index now works, as it was previously removed in #6697 to address the issue of incorrect childname index.

The primary cause of the initial issue was that:
- `response` was always a 2D array of size `[1][n]`, and
- `childIdx` from the parent is the correct index to use?

Reply:
Yes you're right! The function `createFormattedDataForOneField` only creates the data for each field, so this meant that previously it did not track the index of the child. This is why we now pass in `childIdx` from the parent.